packages/miniflare/src/plugins/hyperdrive/index.ts

```diff
@@ -71,13 +71,21 @@ export const HyperdriveSchema = z
 		return {
 			database: url.pathname.replace("/", ""),
 			user: url.username,
-			password: url.password,
+			password: decodeIfURLEncoded(url.password),
 			scheme: url.protocol.replace(":", ""),
 			host: url.hostname,
 			port: port,
 		};
 	});
 
+function decodeIfURLEncoded(str: string): string {
+	try {
+		return decodeURIComponent(str);
+	} catch (e) {
+		return str;
+	}
+}
+
```

Comment:
Will this ever throw? I think anything that would cause `decodeURIComponent` to throw would also cause `new URL` to throw on line 11